src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -541,9 +689,778 @@ private bool CheckSensorHandle()
+                    return true;
+                };
+
+                int error = Interop.SensorRecoder.RecorderRead(type, recorderQuery, _sensorRecorderDataCallBack, IntPtr.Zero);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in asynchronously reading recorder");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderRead Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Starts to record a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to be recorded.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderStart(int type)
+        {
+            if (CheckRecorderOption()) { 
+                int error = Interop.SensorRecoder.RecorderStart(type, recorderOption);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in starting recorder");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderStart Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder option is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Stops recording a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to be recorded.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void RecorderStop(int type)
+        {
+            int error = Interop.SensorRecoder.RecorderStop(type);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in starting recorder");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.RecorderStop Failed");
+            }
+
+        }
+
+        /// <summary>
+        ///  This function creates a sensor provider handle with a given URI.
+        /// </summary>
+        /// <param name="uri">
+        /// The URI of sensor to be created.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        private void CreateProvider(string uri)
+        {
+            int error = Interop.SensorProvider.CreateProvider(uri, out _sensorProviderHandle);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in creating sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.CreateProvider Failed");
+            }
+        }
+
+        /// <summary>
+        /// Registers the sensor provider.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void AddProvider()
```

Comment:
You need to provide a parameter because the method name contain `Add`.

Reply:
done.